File: greedy/179. Largest Number.js
`Given a list of non-negative integers nums, arrange them such that they form the largest number and return it.

Since the result may be very large, so you need to return a string instead of an integer.

 

Example 1:

Input: nums = [10,2]
Output: "210"
Example 2:

Input: nums = [3,30,34,5,9]
Output: "9534330"`;

const largestNumber = (nums) => {
  nums = nums.map((num) => num.toString());
  const compare = (a, b) => (a + b - (b + a) > 0 ? -1 : 1);

  nums.sort((a, b) => compare(a, b));
  return String(parseInt(nums.join("")));
};
console.log(largestNumber([10, 2]));

File: soting/179.largest number.py

from typing import List
class Solution:
    def largestNumber(self, nums: List[int]) -> str:
        # convert to string
        nums=list(map(str,nums))
        # sort the list
        nums.sort(key=lambda x:x*10,reverse=True)

        return str(int(''.join(nums)))


if __name__ == '__main__':
    print("TESTING THE LARGEST NUMBER")
    s=Solution()
    print(s.largestNumber([10,2])) # 210
    print(s.largestNumber([3,30,34,5,9])) # 9534330
    print(s.largestNumber([1])) # 1
    print(s.largestNumber([0,0,0])) # 10
    print("END OF TESTIN")
        

        
